caddyhttp/fastcgi/setup.go

```diff
@@ -147,13 +159,28 @@ func fastcgiParse(c *caddy.Controller) ([]Rule, error) {
 			}
 		}
 
-		rule.balancer = &roundRobin{addresses: upstreams, index: -1}
+		if srvUpstream {
+			balancer, err := parseSRV(upstreams[0])
+			if err != nil {
+				return rules, c.Err("malformed service locator string: " + err.Error())
+			}
+			rule.balancer = balancer
+		} else {
+			rule.balancer = &roundRobin{addresses: upstreams, index: -1}
+		}
 
 		rules = append(rules, rule)
 	}
 	return rules, nil
 }
 
+func parseSRV(locator string) (*srv, error) {
```

Comment:
a panic is bound to happen when the `len(location) <= 6`, a check could be for that and an error can be returned.

Reply:
It will not panic because control will only reach `parseSRV` if the check on line #73 guarantees at least 6 characters and those will be `srv://`. I however added a check to raise an error if the locator consists of only those 6 characters and nothing else.